packages/editor/src/store/reducer.js

```diff
@@ -913,6 +913,29 @@ export function postLock( state = { isLocked: false }, action ) {
 	return state;
 }
 
+/**
+ * Post saving lock.
+ *
+ * When post saving is locked, the post cannot be published or updated.
+ *
+ * @param {PostSavingLockState} state  Current state.
+ * @param {Object}              action Dispatched action.
+ *
+ * @return {PostLockState} Updated state.
+ */
+export function postSavingLock( state = {}, action ) {
+	const { lockName } = action;
```

Comment:
Minor: Since a reducer is called on every single action, it's generally best to avoid doing anything until you're certain to be working with a relevant action. On its own, the allocation here is not so concerning, but an unfamiliar future maintainer may perceive this to imply that logic targeting the lock-specific handling is okay to occur prior to switch case, and feel comfortable to make enhancements that at best harm performance and at worst introduce bugs.

tl;dr Best to move into the `case` handler(s).

Reply:
Adjusted in https://github.com/WordPress/gutenberg/pull/10649/commits/50d5a49c906e9c724f0fc06644fdf323a9f1ace1